### ISSUE TITLE
0 tables - all the time - when generating migration (in neon example deployment) [inside pnpm monorepo]

### ISSUE THREAD

**Issue by sinitsa** [drizzle-team/drizzle-kit-mirror#24]
Hello!

After installing the neon-cf example (on Mac), I first had to update all packages, then run the drizzle-kit up:pg (successful), after which I managed to upload the migration to Neon, and play some with worker insert/select - it works.

But when i try to change something in schema and run drizzle-kit generate:pg- seems like no schemes are found, just a migration is created with drop users only: /
0 tables
[✓] Your SQL migration file ➜ drizzle/0001_blue_mongu.sql 🚀

Why??

It all happens in a pnpm monorep - maybe problem depends? I already tried to install it in the root, change paths, etc. - anyway i got the message - 0 tables  ??

Later, i try to change paths to totally invalid - anyway all the time i got:

# pn drizzle-kit check:pg
drizzle-kit: v0.17.1
drizzle-orm: v0.23.2

Reading drizzle.config.json
Everything's fine 🐶🔥

# pn drizzle-kit generate:pg
drizzle-kit: v0.17.1
drizzle-orm: v0.23.2

0 tables

No schema changes, nothing to migrate 😴
......

What i'm doing wrong??

**Comment by AndriiSherman** [comment]
It depends on several factors(Not all of those your fault. We will try to see where we can `warn` developers more on something they may be doing wrong)

1. Could you please check, that you have setup a path to schema files correctly?
2. Could please check if `out` variable for migrations folder was setup correctly?
I know it should, but just to double check

3. Can you please show what is inside `mets/_journal.json` file?

This info may help me a bit more. Thanks!

Also if you want you can setup simple project for it or share access with me so I could find more details on error and see what you can do to solve this issue and what we can improve in drizzle-kit suggestions flow

**Comment by sinitsa** [comment]
I tried to make a separate repository - and everything worked right away. Then I tried again the same thing inside the monorepo - again the same problem - it looks like the problem is related to the work of the drizzle-kit inside the monorepo (pnpm in my case)

% pn dkit:up

> drizzle-neon-bug@0.0.0 dkit:up /Users/havok/Work/flute/packages/drizzle-neon-clean
> drizzle-kit up:pg

drizzle-kit: v0.17.1
drizzle-orm: v0.23.2

Reading drizzle.config.json
drizzle/20230203200203/snapshot.json
[✓] drizzle/20230203200203/snapshot.json
Everything's fine 🐶🔥


% dkit:generate

> drizzle-neon-bug@0.0.0 dkit:generate /Users/havok/Work/flute/packages/drizzle-neon-clean
> drizzle-kit generate:pg

drizzle-kit: v0.17.1
drizzle-orm: v0.23.2

0 tables

[✓] Your SQL migration file ➜ drizzle/0001_perpetual_luminals.sql 🚀

**Comment by rajatbarman** [comment]
I had the same issue as I wasn't using the right schema path, it worked after correction.
I second @sinitsa and congratulate you for creating this excellent tool.
One suggestion is to have CLI throw out an error on invalid path for `--schema`

**Comment by sinitsa** [comment]
I didn’t quite understand about the drizzle-kit, is it not open source yet?

**Comment by AndriiSherman** [comment]
@sinitsa if you will remove migration folders in you monorepo and then regenerate it? Would that still not work?
I mean whole folder, where sql files, meta and journal are?

**Comment by AndriiSherman** [comment]
drizzle-orm is fully open-sourced
drizzle-kit is partially open-sourced(as long as code is still available from npm) and will be fully open sourced soon

**Comment by sinitsa** [comment]
> @sinitsa if you will remove migration folders in you monorepo and then regenerate it? Would that still not work? I mean whole folder, where sql files, meta and journal are?

yes, or i can just change "out" to a new name - it makes "empty" _journal.json inside:
{"version":"5","dialect":"pg","entries":[]}

inside monorepo ... in the separate repo - everything ok

**Comment by AndriiSherman** [comment]
@sinitsa seems like we may found 1 of a reasons why journal is empty
Will create patch release today and ping you here

**Comment by sinitsa** [comment]
@AndriiSherman I'm sorry, not done yet? I tried updating to the latest versions - but it seems the same problem

**Comment by michaelgmcd** [comment]
Make sure your drizzle-orm version is the same across the monorepo. Run this from the root to make sure: `pnpm ls -r --depth=10 drizzle-orm`

**Comment by ledburyb** [comment]
In case anyone else stumbles across this like I did, I had the same issue with generate finding 0 tables inside a pnpm monorepo. I don't know why but from the Discord (https://discord.com/channels/1043890932593987624/1090530481579229224) I found the solution is to make sure that drizzle-orm is only installed in a single package inside the monorepo.

**Comment by AdditionAddict** [comment]
Had same issue but needed to run `pnpm i` in new `cmd` and things started working.

**Comment by sinitsa** [comment]
hmm cool... 2.5 months, nothing new..

**Comment by crushingCodes** [comment]
I have a monorepo using npm and still had the same problem. 
To get it to work I ended up running `npm i drizzle-kit`

then running:
`node node_modules/drizzle-kit/index.js generate:pg`
`// 9 tables`


This is the tree for `./drizzle` which is a subfolder in my monorepo
```
.
├── drizzle.config.ts
├── migrate.ts
├── migrations
│   ├── 0000_mysterious_echo.sql
│   ├── 0001_equal_angel.sql
│   └── meta
├── node_modules
...
│   ├── drizzle-kit
...
├── package-lock.json
├── package.json
└── schema.ts
```

**Comment by lika85456** [comment]
Same problem here using turborepo. Tried using `schema: "../../**/*.schema.ts",` to find all schemas in monorepo. Drizzle kit finds all of them, but for schemas outside the main app 0 tables are loaded.

**Comment by t0mri** [comment]
same issue here with pg.

my file structure:
│
├──db
│   ├── schema.ts
│   └── db.ts

```typescript
// schema.ts

import { serial, text, timestamp, pgTable } from "drizzle-orm/pg-core";

export const user = pgTable("user", {
  id: serial("id"),
  name: text("name"),
  email: text("email"),
  password: text("password"),
  role: text("role").$type<"admin" | "customer">(),
  createdAt: timestamp("created_at"),
  updatedAt: timestamp("updated_at"),
});
```
```typescript
// drizzle.config.ts:

import type { Config } from "drizzle-kit";

export default {
  schema: "./db/schema.ts",
  out: "./db",
} satisfies Config;
```
```typescript
// package.json

 "scripts": {
    "generate": "drizzle-kit generate:pg"
  },
"devDependencies": {
    "@types/pg": "^8.10.5",
    "bun-types": "latest",
    "drizzle-kit": "^0.19.13"
  },
```
```typescript
// terminal

tomri:/mnt/t/tomri/Dev/auth $ bun run generate
$ drizzle-kit generate:pg
drizzle-kit: v0.19.13
drizzle-orm: v0.28.6

No config path provided, using default 'drizzle.config.ts'
Reading config file '/mnt/t/tomri/Dev/auth/drizzle.config.ts'
0 tables


No schema changes, nothing to migrate 😴
```
